mlton/front-end/front-end.fun

```diff
@@ -40,6 +40,30 @@ fun lexAndParse (source: Source.t, ins: In.t): Ast.Program.t =
                Ast.Program.T []
             end
       val () = Ast.Program.checkSyntax result
+
+      (* Outputs AST to a file if Control.keepAST is true *)
+      val () =
+        if !Control.keepAST
+          then
+            let
+               val inputFile = File.toString (!Control.inputFile)
+               val outputFile = concat [inputFile, ".ast"]
```

Comment:
Using `concat` matches with other uses of building file names from `Control.inputFile`.